File: test-cases/src/user_service.py
users = []


def register_user(name, email):
    if not name or not email:
        raise ValueError("Name and email are required")

    if "@" not in email:
        raise ValueError("Invalid email format")

    user = {
        "id": len(users) + 1,
        "name": name,
        "email": email,
    }

    users.append(user)
    return user


def get_users():
    return users


def find_user_by_email(email):
    for u in users:
        if u["email"] == email:
            return u
    return None


def clear_users():
    users.clear()

File: test-cases/src/userService.js
const users = [];

function registerUser(name, email) {
  if (!name || !email) {
    throw new Error("Name and email are required");
  }

  if (!email.includes("@")) {
    throw new Error("Invalid email format");
  }

  const user = {
    id: users.length + 1,
    name,
    email,
  };

  users.push(user);
  return user;
}

function getUsers() {
  return users;
}

function findUserByEmail(email) {
  return users.find((u) => u.email === email);
}

function clearUsers() {
  users.length = 0;
}

module.exports = { registerUser, getUsers, findUserByEmail, clearUsers };
